### ISSUE TITLE
Bootstrapping an already bootstrapped node.

### ISSUE THREAD

**Issue by zalokhan** [CorfuDB/CorfuDB#240]
Started server A on 172.17.0.2:9000: 
`./corfu_server -ms -a 172.17.0.2 9000`

I encountered this bug while trying to bootstrap an already bootstrapped node with a different layout.

Started server B on 172.17.0.3:9000: 
`./corfu_server -ms -a 172.17.0.3 9000`

I have started both these servers with the –s flag as bootstrapped single nodes.
I attempted to add node B to node A’s layout.
`./corfu_layouts add_layout -c 172.17.0.2:9000 -e 172.17.0.3:9000`

Ideally it should have rejected this.
Actual output at B:

```
05:28:40.609 WARN  [event-1] o.c.i.LayoutServer - Got a request to bootstrap a server which is already bootstrapped, rejecting!
05:28:40.618 INFO  [event-1] o.c.i.LayoutServer - Received SET_EPOCH, moving to new epoch 1
05:28:41.026 ERROR [event-1] o.c.i.NettyServerRouter - Error in handling inbound message, {}
java.io.IOException: Connection reset by peer
                at sun.nio.ch.FileDispatcherImpl.read0(Native Method)

```

So as we see although it gave this warning, it accepted the epoch with the higher rank and got bootstrapped with the new layout.

This is the layout after adding B to A's layout.

```
root@98dc02452c80:/Corfu/bin# ./corfu_layouts query -c 172.17.0.3:9000
05:28:55.854 INFO  [main] o.c.r.CorfuRuntime - Connecting to Corfu server instance, layout servers=[172.17.0.3:9000]
RESPONSE:
{
  "layoutServers": [
    "172.17.0.2:9000",
    "172.17.0.3:9000"
  ],
  "sequencers": [
    "172.17.0.2:9000"
  ],
  "segments": [
    {
      "replicationMode": "CHAIN_REPLICATION",
      "start": 0,
      "end": -1,
      "stripes": [
        {
          "logServers": [
            "172.17.0.2:9000"
          ]
        }
      ]
    }
  ],
  "epoch": 1
}
```

**Comment by rogermichoud** [comment]
Is this still considered as a bug ? Now that we want the ability to add a new node after being bootstrapped, isn't this the expected behavior ?

**Comment by zalokhan** [comment]
Yes this is still a bug. The issue now is that paxos should also take care of a cluster ID.
If a paxos round runs in cluster A adding a member 'X' of cluster B. X should reject this prepare and propose of the paxos since it is already committed to an existing cluster ID (cluster B).

With the new cmdlets you can reproduce this issue as follows:

Set up 2 servers:
`bin/corfu_server -ms 9000`
`bin/corfu_server -ms 9001`
Note: Both are in single mode, hence already bootstrapped.

Now edit the layout of the 1st server (9000) to add 9001.
Expected result: 9001 should reject this layout proposal.
Actual result: 9001 accepts this new layout proposal and both 9000 and 9001 have the same layout.

**Comment by zalokhan** [comment]
I am working on this. I'll close this with the implementation of the cluster ID.

**Comment by slfritchie** [comment]
Mid-June issues review: ping @zalokhan

**Comment by zalokhan** [comment]
Waiting for review of #534

**Comment by zalokhan** [comment]
Resolved by #1111
